include/color_spinor_field.h

```diff
@@ -426,6 +426,12 @@ namespace quda
      */
     ColorSpinorField &operator=(ColorSpinorField &&field);
 
+    /**
+       @brief Returns if the object is empty (not initialized)
+       @return true if the object has not been allocated, otherwise false
+    */
+    bool empty() const { return !init; }
```

Comment:
Perhaps outside the scope of this PR, but can the variable `init`, and by extension this function, become a member of the parent `LatticeField` class?

Reply:
I did start to look into it when I wrote it, but it wasn't quite immediately trivial.  Good suggestion, but I think that should be for later PR.